ebpf-fw/src/main.rs

```diff
@@ -0,0 +1,61 @@
+use aya::programs::{tc, SchedClassifier, TcAttachType};
+use aya::{include_bytes_aligned, Bpf};
+use aya_log::BpfLogger;
+use clap::Parser;
+use log::{info, warn, debug};
+use tokio::signal;
+
+#[derive(Debug, Parser)]
+struct Opt {
+    #[clap(short, long, default_value = "lo")]
+    iface: String,
+    #[clap(long)]
+    log: bool,
+}
+
+#[tokio::main]
+async fn main() -> Result<(), anyhow::Error> {
+    let opt = Opt::parse();
+
+    env_logger::init();
+
+    // Bump the memlock rlimit. This is needed for older kernels that don't use the
+    // new memcg based accounting, see https://lwn.net/Articles/837122/
+    let rlim = libc::rlimit {
```

Comment:
Do we actually need this "older kernels" (which version?) backward compatibility support? The link does not present exact kernel versions but our NixOS dev and target environments sport `6.x` kernels on `x86_64`.
And `aarch64` support for possible `5.x` kernels we should separately anyway.

This is not just nit-picking but a practical consideration - when we don't need it (bc support), let's remove it. We get rid of `unsafe` boilerplate code in the process.